File: Assignment_49X/Assignment_49.1.cpp
#include <iostream>
using namespace std;

struct node
{
    int data;
    struct node *next;
};

typedef struct node NODE;
typedef struct node *PNODE;

class SinglyLL
{
public:
    PNODE first;
    int iCount;

public:
    SinglyLL()
    {
        cout << "Object of SinglyLL gets created" << "\n";
        first = NULL;
        iCount = 0;
    }

    void InsertFirst(int no)
    {
        PNODE newn = NULL;

        newn = new NODE;

        newn->data = no;
        newn->next = NULL;

        if (first == NULL)
        {
            first = newn;
        }
        else
        {
            newn->next = first;
            first = newn;
        }
    }

    void Display()
    {
        PNODE temp = NULL;

        temp = first;

        while (temp != NULL)
        {
            cout << "| " << temp->data << " |->";
            temp = temp->next;
        }

        cout << "NULL\n";
    }

    int Difference()
    {   
        PNODE temp = NULL;

        temp = first;
        int iMax = temp->data;
        int iMin = temp->data;

        while(temp != NULL)
        {
            if((temp->data) > iMax)
            {
                iMax = temp->data; 
            }

            if((temp->data) < iMin)
            {
                iMin = temp->data; 
            }
            temp = temp->next;
        }
        
        return (iMax - iMin);
    }
};

int main()
{
    SinglyLL obj;
    int iRet = 0;

    obj.InsertFirst(51);
    obj.InsertFirst(40);
    obj.InsertFirst(31);
    obj.InsertFirst(20);
    obj.InsertFirst(11);
    obj.InsertFirst(31);

    obj.Display();

    iRet = obj.Difference();

    cout<<"Difference is:"<<iRet<<"\n";

    return 0;
}

File: Assignment_49XX/Assignment_49_1.java


class node
{
    public int data;
    public node next;
    
    public node(int no)
    {
        data = no;
        next = null;
    }
}

class SinglyLL
{
    private node first;
    private int iCount;

    public SinglyLL() 
    {
        System.out.println("Object Of SinglyLL gets created");
        first = null;
        iCount = 0;
    }

    public void InsertFirst(int no)
    {
        node newn  = null;

        newn = new node(no);

        if(first == null)
        {
            first = newn;
        }
        else
        {
            newn.next = first;
            first = newn;
        }
    }
    public void Display()
    {
        node temp = null;

        temp = first;

        while(temp != null)
        {
            System.out.print("| "+temp.data+" |->");
            temp = temp.next;
        }

        System.out.println("null\n");
    }

    
    public int Difference()
    {
        node temp = first;

        int iMax = temp.data;
        int iMini = temp.data;


        while(temp != null)
        {   
            if((temp.data) > iMax)
            {
                iMax = temp.data;
            }

            if((temp.data) < iMini)
            {
                iMini = temp.data;
            }
            temp = temp.next;
             
        }
       return (iMax - iMini);
    }
}

class Assignment_49_1
{
    public static void main(String Arr[])
    {
        SinglyLL obj;
        int iRet = 0;

        obj = new SinglyLL();

        obj.InsertFirst(51);
        obj.InsertFirst(41);
        obj.InsertFirst(28);
        obj.InsertFirst(6);
        obj.InsertFirst(10);
        obj.InsertFirst(2);

        obj.Display();

        iRet = obj.Difference();

        System.out.println("The Difference is: "+iRet);

        obj = null;

        System.gc();
    }
}